stratum/hal/lib/common/common.proto

```diff
@@ -692,10 +692,89 @@ message TofinoConfig {
     repeated DropTarget drop_targets = 1;
   }
 
+  // QoS configuration for the traffic manager.
+  message TofinoQoSConfig {
+    enum ApplicationPool {
+      UNKNOWN_APP_POOL = 0;
+      INGRESS_APP_POOL_0 = 1;
+      INGRESS_APP_POOL_1 = 2;
+      INGRESS_APP_POOL_2 = 3;
+      INGRESS_APP_POOL_3 = 4;
+      EGRESS_APP_POOL_0 = 5;
+      EGRESS_APP_POOL_1 = 6;
+      EGRESS_APP_POOL_2 = 7;
+      EGRESS_APP_POOL_3 = 8;
+    }
+    enum Baf {
+      UNKNOWN_BAF = 0;
+      BAF_1_POINT_5_PERCENT = 1;
+      BAF_3_PERCENT = 2;
+      BAF_6_PERCENT = 3;
+      BAF_11_PERCENT = 4;
+      BAF_20_PERCENT = 5;
+      BAF_33_PERCENT = 6;
+      BAF_50_PERCENT = 7;
+      BAF_66_PERCENT = 8;
+      BAF_80_PERCENT = 9;
+      DISABLE_BAF = 10;
+    }
+    enum SchedulingPriority {
+      UNKNOWN_PRIO = 0;
+      PRIO_0 = 1;
+      PRIO_1 = 2;
+      PRIO_2 = 3;
+      PRIO_3 = 4;
+      PRIO_4 = 5;
+      PRIO_5 = 6;
+      PRIO_6 = 7;
+      PRIO_7 = 8;
+    }
+    message PoolConfig {
+      ApplicationPool pool = 1;  // required
+      uint32 pool_size = 2;
+      bool enable_color_drop = 3;
+    }
+    message PpgConfig {
+      uint32 sdk_port = 1;  // required
```

Comment:
consider supporting singleton port ids

Reply:
Since we currently pass the entire config to the SdeWrapper, which has no knowledge about singleton ports, supporting those is infeasible.